src/dmd/statementsem.d

```diff
@@ -2262,6 +2265,10 @@ else
         CtorFlow ctorflow_root = scd.ctorflow.clone();
 
         ifs.ifbody = ifs.ifbody.semanticNoScope(scd);
+        if (scd.has_label) {
+            ifs.ifbody.has_label = true;
```

Comment:
This might actually not be better to set `ss.has_label` in `visit(ScopeStatement)` (and maybe test here on the `if.body`).